crates/worker/src/partition/snapshots/repository.rs

```diff
@@ -428,13 +428,16 @@ impl SnapshotRepository {
                         .map_err(|e| anyhow!("Failed to download snapshot file {:?}: {}", key, e))?
                         .into_stream(),
                 );
+
                 let mut snapshot_file =
                     tokio::fs::File::create_new(&file_path).await.map_err(|e| {
                         anyhow!("Failed to create snapshot file {:?}: {}", file_path, e)
                     })?;
                 let size = io::copy(&mut file_data, &mut snapshot_file)
                     .await
                     .map_err(|e| anyhow!("Failed to download snapshot file {:?}: {}", key, e))?;
+                snapshot_file.shutdown().await?;
```

Comment:
This might have been a production bug that could have caused files to be truncated on download! Thanks for catching this, @muhamadazmy 🙏